js-commit-guidelines.md

```diff
@@ -0,0 +1,99 @@
+# Git Commit Guidelines
+
+We have very precise rules over how our git commit messages can be formatted.
+This leads to **more readable messages** that are easy to follow when
+looking through the **project history**.  But also,
+we use the git commit messages to **generate the change log**.
```

Comment:
This is not written by me and left unchanged as much as possible from the angular commit convention document.